R/ebird_conn.R

```diff
@@ -1,52 +1,83 @@
-
-#' A `[DBI]`-style database connection to the imported eBird data
+#' Set up a `DBI`-style database connection to the imported eBird data
 #' 
-#' Returns a database connection to the local eBird dataset.
-#' @export
+#' Parquet files can be accessed as though they were relational database tables 
+#' by setting up a view to the file using DuckDB. This function sets up a view 
+#' on either the checklist or observation dataset and returns a [DBI]-style 
+#' database connection to the data. The returned object can then be queried 
+#' with SQL syntax via [DBI] or with [dplyr] syntax via [dbplyr]. For the latter 
+#' approach, consider using the [checklists()] and [observations()] functions 
+#' which will return [tbl] objects ready for access using [dplyr] syntax.
+#' 
+#' @param dataset the type of dataset to set up a connection to, either the 
+#'   observations of checklists.
+#' @param memory_limit the memory limit for DuckDB.
 #' 
+#' @return A [DBI] connection object using to communicate with the DuckDB 
+#'   database containing the eBird data.
+#' @export
 #' @examples
-#' # tempdir settings are just for testing! Don't copy this.
-#' Sys.setenv("BIRDDB_HOME"=tempdir())
+#' # only use a tempdir for this example, don't copy this for real data
+#' temp_dir <- file.path(tempdir(), "birddb")
+#' Sys.setenv("BIRDDB_HOME" = temp_dir)
 #' 
-#' tar <- ebird_sample_data()
+#' # get the path to a sample dataset provided with the package
+#' tar <- sample_observation_data()
+#' # import the sample dataset to parquet
 #' import_ebird(tar)
-#' con <- ebird_conn()
+#' # set up the database connection
+#' con <- ebird_conn(dataset = "observations")
 #' 
-ebird_conn <- function() {
-  
-  tblname <- "ebd"
-  parquet <- ebird_parquet_files()
+#' unlink(temp_dir, recursive = TRUE)
+ebird_conn <- function(dataset = c("observations", "checklists"), 
+                       memory_limit = 4) {
```

Comment:
I've put this as a parameter here defaulting to 4 GB. Not sure if this is the best default or if there's a better way to pass this, e.g. env variable. Regardless, it seems clear we need to provide some default limit to avoid crashing R.

Reply:
For now this seems reasonable, so these are really just a few thoughts for future reference:

Ultimately I don't think it should crash R not to set PRAGMA limits -- `duckdb` is supposed to be able to detect free RAM, but I will try and explore that a little further and see if we can follow up with the `duckdb` team on this.  If it does wind up being pretty necessary, I think it would be best to make it possible to set this 'globally', e.g. with a variable in a .Renviron, but also allow the user to override it.  Last thought: it's possible it would be more natural to include units on this, e.g. as a character vector "4 GB" instead of an integer, but might then need some assertions/error-checks that the user-supplied term is parseable.  (`fs::as_fs_bytes()` may provide some inspiration)

Reply:
ok, this all sounds good! I think for now I'll just leave things simple for the sake of testing, but hopefully they can get this resolved on the `duckdb` side so it does the "right" thing with memory out of the box.